packages/twenty-front/src/modules/workflow/workflow-diagram/workflow-edges/components/WorkflowDiagramIteratorCompletedEmptyTriggerEdgeEditable.tsx

```diff
@@ -0,0 +1,182 @@
+import { useEdgeState } from '@/workflow/workflow-diagram/workflow-edges/hooks/useEdgeState';
+import { useCreateStep } from '@/workflow/workflow-steps/hooks/useCreateStep';
+import { useDeleteEdge } from '@/workflow/workflow-steps/hooks/useDeleteEdge';
+import { useIsFeatureEnabled } from '@/workspace/hooks/useIsFeatureEnabled';
+import styled from '@emotion/styled';
+import { t } from '@lingui/core/macro';
+import {
+  EdgeLabelRenderer,
+  type EdgeProps,
+  getBezierPath,
+} from '@xyflow/react';
+import { type MouseEvent, useContext } from 'react';
+import { useSetRecoilState } from 'recoil';
+import { isDefined } from 'twenty-shared/utils';
+import { IconFilter, IconPlus, IconTrash } from 'twenty-ui/display';
+import { FeatureFlagKey } from '~/generated/graphql';
+
+type WorkflowDiagramIteratorCompletedEmptyTriggerEdgeEditableProps =
+  EdgeProps<WorkflowDiagramEdge>;
+
+const StyledTopLabelContainer = styled.div<{
+  sourceX: number;
+  sourceY: number;
+}>`
+  position: absolute;
+  transform: translate(-50%, 0%)
+    translate(
+      ${({ sourceX, sourceY }) => `${sourceX}px, ${Math.floor(sourceY)}px`}
+    )
+    translateY(10px);
+  width: fit-content;
+`;
+
+export const WorkflowDiagramIteratorCompletedEmptyTriggerEdgeEditable = ({
+  source,
+  target,
+  sourceX,
+  sourceY,
+  targetX,
+  targetY,
+  markerStart,
+  markerEnd,
+}: WorkflowDiagramIteratorCompletedEmptyTriggerEdgeEditableProps) => {
+  const isWorkflowBranchEnabled = useIsFeatureEnabled(
+    FeatureFlagKey.IS_WORKFLOW_BRANCH_ENABLED,
+  );
+
+  const { isInRightDrawer } = useContext(ActionMenuContext);
+
+  const { isEdgeHovered } = useEdgeState();
+
+  const [edgePath, labelX, labelY] = getBezierPath({
+    sourceX,
+    sourceY,
+    targetX,
+    targetY,
+  });
+
+  const workflowVisualizerWorkflowId = useRecoilComponentValue(
+    workflowVisualizerWorkflowIdComponentState,
+  );
+  const workflow = useWorkflowWithCurrentVersion(workflowVisualizerWorkflowId);
+
+  const { createStep } = useCreateStep({ workflow });
+
+  const { deleteEdge } = useDeleteEdge();
+
+  const { startNodeCreation, isNodeCreationStarted } = useStartNodeCreation();
+
+  const nodeCreationStarted = isNodeCreationStarted({
+    parentStepId: source,
+    nextStepId: target,
+  });
+
+  const setCommandMenuNavigationStack = useSetRecoilState(
+    commandMenuNavigationStackState,
+  );
+
+  const { openWorkflowEditFilterInCommandMenu } =
+    useOpenWorkflowEditFilterInCommandMenu();
+
+  const handleCreateFilter = async () => {
+    const createdStep = await createStep({
+      newStepType: 'FILTER',
+      parentStepId: source,
+      nextStepId: target,
+    });
+
+    if (!isDefined(createdStep)) {
+      return;
+    }
+
+    if (!isInRightDrawer) {
+      setCommandMenuNavigationStack([]);
+    }
+
+    openWorkflowEditFilterInCommandMenu({
+      stepId: createdStep.id,
+      stepName: createdStep.name,
+    });
+  };
```

Comment:
We will be able to simplify this code as it's 1-1 duplicated on several components. I plan to do it later.